test/engine/SpatialJoinTest.cpp

```diff
@@ -0,0 +1,1957 @@
+  std::vector<std::string> rows(expected_output->size(), "");
+  // std::vector<std::string> tableToTest = printTable(qec, resultTableToTest);
+  for (size_t i = 0; i < expected_output->size(); i++) {
+    rows.at(i) = tableToTest.at(i);  // initialize as copy, reorder when needed
+  }
+  for (size_t i = 0; i < expected_output->size(); i++) {
+    for (size_t k = 0; k < tableToTest.size(); k++) {
+      if (tableToTest.at(k) == expected_output->at(i)) {
+        rows.at(i) = tableToTest.at(k);  // change order of expected output
+        break;
+      }
+    }
+  }
+  // if all rows from the expected output are in the result table and the tables
+  // are equally large, then both tables are equal (assuming each row is unique)
+  ASSERT_EQ(tableToTest.size(), expected_output->size());
+  for (size_t i = 0; i < expected_output->size(); i++) {
+    ASSERT_EQ(expected_output->at(i), rows.at(i));
+  }
+}
+
+// create a small test dataset, which focuses on points as geometry objects
+std::string createSmallDatasetWithPoints() {
+  // note, that some of these objects have a polygon representation, but for
+  // testing purposes, they get represented as a point here. I took those
+  // points, such that it is obvious, which pair of objects should be included,
+  // when the maximum distance is x meters. Please note, that these datapoints
+  // are not copied from a real input file. Copying the query will therefore
+  // likely not result in the same results as here (also the names, coordinates,
+  // etc. might be different in the real datasets)
+
+  // Uni Freiburg TF, building 101
+  std::string kg = "<node_1> <name> \"Uni Freiburg TF\" .";
+  kg += "<node_1> <hasGeometry> <geometry1> .";
+  kg += "<geometry1> <asWKT> \"POINT(7.83505 48.01267)\" .";
+  // Minster Freiburg
+  kg += "<node_2> <name> \"Minster Freiburg\" .";
+  kg += "<node_2> <hasGeometry> <geometry2> .";
+  kg += "<geometry2> <asWKT> \"POINT(7.85298 47.99557)\" .";
+  // London Eye
+  kg += "<node_3> <name> \"London Eye\" .";
+  kg += "<node_3> <hasGeometry> <geometry3> .";
+  kg += "<geometry3> <asWKT> \"POINT(-0.11957 51.50333)\" .";
+  // statue of liberty
+  kg += "<node_4> <name> \"Statue of liberty\" .";
+  kg += "<node_4> <hasGeometry> <geometry4> .";
+  kg += "<geometry4> <asWKT> \"POINT(-74.04454 40.68925)\" .";
+  // eiffel tower
+  kg += "<node_5> <name> \"eiffel tower\" .";
+  kg += "<node_5> <hasGeometry> <geometry5> .";
+  kg += "<geometry5> <asWKT> \"POINT(2.29451 48.85825)\" .";
+  return kg;
+}
+
+// this function creates an input as a test set and returns it
+std::string createTestKnowledgeGraph(bool verbose) {
+  auto addPoint = [](string* kg, double lon, double lat) {
+    string name =
+        "Point_" + std::to_string(lon) + "_" + std::to_string(lat) + "";
+    *kg += "";
+
+    *kg += name;
+    *kg += " geo:asWKT Point(";
+    *kg += std::to_string(lon);
+    *kg += " ";
+    *kg += std::to_string(lat);
+    *kg += ")^^geo:wktLiteral .\n";
```

Comment:
You can factor out the creation of `\"POINT(..., ...)\"^^geo::wktLiteral` from two doubles into a separate function.